lke-configurations/alerts.cypher

```diff
@@ -1,56 +1,137 @@
-// /*
-//  * Name: UNNAMED #1
-//  * Desc: Find Mortgage Loans with a Realtor or a broker in a different state
-//  *
-//  * Target: m
-//  *
-//  * Custom columns:
-//  *   name   | Name    | text
-//  *   b.name | Broker  | text
-//  *   r.name | Realtor | text
-//  */
-
-MATCH (b:Company)<-[hb:HAS_BROKER]-(m:MortgageLoan)<-[hl:HAS_LOAN]-(p)
-MATCH (m)-[hr:HAS_REALTOR]->(r:Company)
-WITH m, p, b, r, hl, hb, hr, split(m.address, ",") as ma, split(b.address, ",") as ba, split(r.address, ",") as ra
-WHERE ma[size(ma)-1] <> ba[size(ba)-1] OR ma[size(ma)-1] <> ba[size(ra)-1]
-RETURN m, p, b, r, hl, hb, hr,  coalesce(p.full_name, p.name) as name, b.name, r.name
-
-
-
 // /*
 //  * Name: Common Clients
 //  * Desc: Find group of bank clients (more than 3) involved in Mortgage Loans that share the same Loaners and Brokers
 //  *
 //  * Target: r
 //  *
 //  * Custom columns:
-//  *   b.name | Broker        | text
-//  *   r.name | Realtor       | text
-//  *   len    | N. of clients | number
+//  *   Broker        | b.name | text
+//  *   Realtor       | r.name | text
+//  *   N. of clients | len    | number
 //  */
 
+
+// Model 1
 MATCH (p{is_client:true})-[hl:HAS_LOAN]->(m:MortgageLoan)
 MATCH (r:Company)<-[hr:HAS_REALTOR]-(m)-[hb:HAS_BROKER]->(b:Company)
-WITH r, b, collect(p) as col, collect(hl) + collect(hr) + collect(hb) as rel
+WITH r, b, collect(p) + collect(m) as col, collect(hl) + collect(hr) + collect(hb) as rel
```

Comment:
Are you sure that `collect(p) + collect(m) as col` is not impacting the condition `size(col) > 3`? It seems more a but introduced to add the missing MortgageLoan to the viz rather than an intended change.

As a best practice I would collect paths for the purpose of showing things in the case. e.g.
```cypher
MATCH loans = (p { is_client: true })-[hl:HAS_LOAN]->(m:MortgageLoan)
MATCH loan_context = (r:Company)<-[hr:HAS_REALTOR]-(m)-[hb:HAS_BROKER]->(b:Company)
WITH r, b, count(p) AS clients, collect(loans) + collect(loan_context) AS context
WHERE clients > 3
RETURN r, b, context
```

(usually if there's a single match I would generically call it `path` or `p` if nor already using the letter as variable)

Reply:
ok